src/dosext/mfs/lfn.c

```diff
@@ -683,7 +683,7 @@ static int wildcard_delete(char *fpath, int drive)
 		free(dir);
 		return lfn_error(PATH_NOT_FOUND);
 	}
-	pattern = malloc(strlen(slash + 1) + 1);
+	pattern = strdup(slash + 1);
```

Comment:
I don't think that's right.
`pattern` is likely overwritten
a line below.

Reply:
Yes, the copying of the string data was incidental to the allocation, but certainly it doesn't make things clearer. I'll think up something else!

Reply:
Is this okay?
```diff
-       pattern = malloc(strlen(slash + 1) + 1);
-       name_ufs_to_dos(pattern, slash + 1);
+       pattern = malloc(strlen(&slash[1]) + 1);
+       name_ufs_to_dos(pattern, &slash[1]);
```

Reply:
Maybe, but I am not really sure
what was the problem.
It looks like the same thing!

Reply:
I don't think there was a problem except that Coverity views `malloc(strlen(string + 1))` as a mistake and that the dev probably meant `malloc(strlen(string) + 1)`. I'm quite happy to drop this,as it's not clearer with either of the two suggested changes, and too much effort just to please Coverity!